### ISSUE TITLE
[3.0.18] WAL issues during shutdown

### ISSUE THREAD

**Issue by hrstoyanov** [orientechnologies/orientdb#8839]
### OS: McOS 

@laa , @tglman 
  
When shutting down ODB 3.0.18 (earlier versions too) server with Ctrl-C the below exceptions occur. Just like #8672, this is incredibly nasty issue - the database can no longer be used and I have to rebuild it completely, potentially losing data.

```
2019-04-13 18:00:42:136 INFO  Storage 'plocal:/Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db' is opened under OrientDB distribution : 3.0.18 - Veloce (build 747595e790a081371496f3bb9c57cec395644d82, branch 3.0.x) [OLocalPaginatedStorage]^C
2019-04-13 20:58:36:268 WARNI Received signal: SIGINT [OSignalHandler]
2019-04-13 20:58:36:269 INFO  OrientDB Server is shutting down... [OServer]
2019-04-13 20:58:36:270 INFO  Shutting down listeners: [OServer]
2019-04-13 20:58:36:270 INFO  - ONetworkProtocolBinary /0.0.0.0:2424: [OServer]
2019-04-13 20:58:36:271 INFO  - ONetworkProtocolHttpDb /0.0.0.0:2480: [OServer]
2019-04-13 20:58:36:271 INFO  Shutting down protocols [OServer]
2019-04-13 20:58:36:271 INFO  Shutting down plugins: [OServerPluginManager]
2019-04-13 20:58:36:271 INFO  - studio [OServerPluginManager]
2019-04-13 20:58:36:271 INFO  - teleporter [OServerPluginManager]
2019-04-13 20:58:36:272 INFO  - custom-sql-functions-manager [OServerPluginManager]
2019-04-13 20:58:36:272 INFO  - script-interpreter [OServerPluginManager]
2019-04-13 20:58:36:272 INFO  - etl [OServerPluginManager]
2019-04-13 20:58:36:272 INFO  - neo4j-importer [OServerPluginManager]
2019-04-13 20:58:36:272 INFO  Shutting down databases: [OServer]
2019-04-13 20:58:36:273 INFO  Orient Engine is shutting down... [Orient]
2019-04-13 20:58:36:273 INFO  - shutdown storage: rehabowner.db... [OrientDBDistributed]Error during WAL writing
java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:88)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.newFileChannel(UnixFileSystemProvider.java:177)
        at java.nio.channels.FileChannel.open(FileChannel.java:287)
        at java.nio.channels.FileChannel.open(FileChannel.java:335)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OWALFile.createWriteWALFile(OWALFile.java:38)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog$RecordsWriter.run(OCASDiskWriteAheadLog.java:1856)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Exception during WAL flush
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog.doFlush(OCASDiskWriteAheadLog.java:1498)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog.flush(OCASDiskWriteAheadLog.java:1359)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.makeFullCheckpoint(OAbstractPaginatedStorage.java:4321)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doClose(OAbstractPaginatedStorage.java:4909)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.close(OAbstractPaginatedStorage.java:682)
        at com.orientechnologies.orient.core.storage.OStorageAbstract.shutdown(OStorageAbstract.java:235)
        at com.orientechnologies.orient.core.db.OrientDBEmbedded.internalClose(OrientDBEmbedded.java:536)
        at com.orientechnologies.orient.core.Orient$OShutdownOrientDBInstancesHandler.shutdown(Orient.java:824)
        at com.orientechnologies.orient.core.Orient.shutdown(Orient.java:394)
        at com.orientechnologies.orient.server.OServer.deinit(OServer.java:597)
        at com.orientechnologies.orient.server.OServer.shutdown(OServer.java:516)
        at com.orientechnologies.orient.server.OServerShutdownHook.run(OServerShutdownHook.java:40)
Caused by: java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog$RecordsWriter.run(OCASDiskWriteAheadLog.java:2054)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:88)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.newFileChannel(UnixFileSystemProvider.java:177)
        at java.nio.channels.FileChannel.open(FileChannel.java:287)
        at java.nio.channels.FileChannel.open(FileChannel.java:335)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OWALFile.createWriteWALFile(OWALFile.java:38)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog$RecordsWriter.run(OCASDiskWriteAheadLog.java:1856)
        ... 7 more
Exception in thread 'OrientDB WAL Flush Task'
java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog$RecordsWriter.run(OCASDiskWriteAheadLog.java:2054)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:88)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.newFileChannel(UnixFileSystemProvider.java:177)
        at java.nio.channels.FileChannel.open(FileChannel.java:287)
        at java.nio.channels.FileChannel.open(FileChannel.java:335)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OWALFile.createWriteWALFile(OWALFile.java:38)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog$RecordsWriter.run(OCASDiskWriteAheadLog.java:1856)
        ... 7 more
Exception `2010CB74` in storage `plocal:/Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db`: 3.0.18 - Veloce (build 747595e790a081371496f3bb9c57cec395644d82, branch 3.0.x)
java.lang.IllegalStateException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog.doFlush(OCASDiskWriteAheadLog.java:1501)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog.flush(OCASDiskWriteAheadLog.java:1359)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.makeFullCheckpoint(OAbstractPaginatedStorage.java:4321)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doClose(OAbstractPaginatedStorage.java:4909)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.close(OAbstractPaginatedStorage.java:682)
        at com.orientechnologies.orient.core.storage.OStorageAbstract.shutdown(OStorageAbstract.java:235)
        at com.orientechnologies.orient.core.db.OrientDBEmbedded.internalClose(OrientDBEmbedded.java:536)
        at com.orientechnologies.orient.core.Orient$OShutdownOrientDBInstancesHandler.shutdown(Orient.java:824)
        at com.orientechnologies.orient.core.Orient.shutdown(Orient.java:394)
        at com.orientechnologies.orient.server.OServer.deinit(OServer.java:597)
        at com.orientechnologies.orient.server.OServer.shutdown(OServer.java:516)
        at com.orientechnologies.orient.server.OServerShutdownHook.run(OServerShutdownHook.java:40)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog.doFlush(OCASDiskWriteAheadLog.java:1498)
        ... 11 more
Caused by: java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog$RecordsWriter.run(OCASDiskWriteAheadLog.java:2054)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:88)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.newFileChannel(UnixFileSystemProvider.java:177)
        at java.nio.channels.FileChannel.open(FileChannel.java:287)
        at java.nio.channels.FileChannel.open(FileChannel.java:335)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OWALFile.createWriteWALFile(OWALFile.java:38)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog$RecordsWriter.run(OCASDiskWriteAheadLog.java:1856)
        ... 7 more
-- error on shutdown storage
java.lang.IllegalStateException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog.doFlush(OCASDiskWriteAheadLog.java:1501)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog.flush(OCASDiskWriteAheadLog.java:1359)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.makeFullCheckpoint(OAbstractPaginatedStorage.java:4321)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doClose(OAbstractPaginatedStorage.java:4909)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.close(OAbstractPaginatedStorage.java:682)
        at com.orientechnologies.orient.core.storage.OStorageAbstract.shutdown(OStorageAbstract.java:235)
        at com.orientechnologies.orient.core.db.OrientDBEmbedded.internalClose(OrientDBEmbedded.java:536)
        at com.orientechnologies.orient.core.Orient$OShutdownOrientDBInstancesHandler.shutdown(Orient.java:824)
        at com.orientechnologies.orient.core.Orient.shutdown(Orient.java:394)
        at com.orientechnologies.orient.server.OServer.deinit(OServer.java:597)
        at com.orientechnologies.orient.server.OServer.shutdown(OServer.java:516)
        at com.orientechnologies.orient.server.OServerShutdownHook.run(OServerShutdownHook.java:40)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog.doFlush(OCASDiskWriteAheadLog.java:1498)
        ... 11 more
Caused by: java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog$RecordsWriter.run(OCASDiskWriteAheadLog.java:2054)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.nio.file.FileAlreadyExistsException: /Users/hristostoyanov/projects/orientdb-3.0.18/databases/rehabowner.db/rehabowner.db.2.wal
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:88)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.newFileChannel(UnixFileSystemProvider.java:177)
        at java.nio.channels.FileChannel.open(FileChannel.java:287)
        at java.nio.channels.FileChannel.open(FileChannel.java:335)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OWALFile.createWriteWALFile(OWALFile.java:38)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.cas.OCASDiskWriteAheadLog$RecordsWriter.run(OCASDiskWriteAheadLog.java:1856)
        ... 7 more

2019-04-13 20:58:36:289 INFO  - shutdown storage: OSystem... [OrientDBDistributed]
2019-04-13 20:58:36:399 INFO  Clearing byte buffer pool [Orient]
2019-04-13 20:58:36:401 INFO  OrientDB Engine shutdown complete [Orient]
2019-04-13 20:58:36:401 INFO  OrientDB Server shutdown complete
 [OServer]%                                                                                                                                                                                         ➜  orientdb-3.0.18 
```

**Comment by laa** [comment]
Hi @hrstoyanov could you replace dot (`.`) in the name of your database by an underscore `_` and try again?

**Comment by hrstoyanov** [comment]
@laa I am testing your suggestion to rename the database file on Win7 and MacOs. So far it seems to work, but let's keep this issue open a little longer until I test more...

**Comment by laa** [comment]
Cool!

**Comment by hrstoyanov** [comment]
@laa,
I concluded that removing the . in the database  name resolved the issue for me. Please document this and decide if you want to close this issue. Thank you!

**Comment by laa** [comment]
Fixed in 3.0.20.
